torch/_tensor.py

```diff
@@ -616,10 +616,12 @@ def __format__(self, format_spec):
             return self.item().__format__(format_spec)
         return object.__format__(self, format_spec)
 
+    @_wrap_type_error_to_not_implemented
     def __ipow__(self, other):  # type: ignore[misc]
         if has_torch_function_variadic(self, other):
             return handle_torch_function(Tensor.__ipow__, (self, other), self, other)
-        return NotImplemented
+        dtype = torch.result_type(self, other)
```

Comment:
I don't think we need `dtype` anymore. 

Also, can we use in-place version of `pow` (although it would be the same as `pow(self, other, out=self)`, it would make more sense for `__ipow__`.

Thanks!